crates/store/src/db/sql/mod.rs

```diff
@@ -1135,6 +1135,68 @@ pub fn unconsumed_network_notes(
     Ok((notes, page))
 }
 
+/// Returns a paginated batch of network notes for a specific account that have not yet been
+/// consumed.
+///
+/// The scope of the query is limited by the specified block number. Notes that have not been
+/// consumed before or within the specified block number are treated as unconsumed.
+///
+/// # Returns
+///
+/// A set of unconsumed network notes with maximum length of `size` and the page to get
+/// the next set.
+pub fn unconsumed_network_notes_for_network_account(
+    transaction: &Transaction,
+    network_account_id_prefix: NetworkAccountPrefix,
+    block_num: BlockNumber,
+    mut page: Page,
+) -> Result<(Vec<NoteRecord>, Page)> {
+    assert_eq!(
+        NoteExecutionMode::Network as u8,
+        0,
+        "Hardcoded execution value must match query"
+    );
+
+    // Select the rowid column so that we can return a pagination token.
+    //
+    // rowid column _must_ come after the note fields so that we don't mess up the
+    // `NoteRecord::from_row` call.
+    let mut stmt = transaction.prepare_cached(&format!(
+        "
+        SELECT {}, rowid
+        FROM notes
+        LEFT JOIN note_scripts ON notes.script_root = note_scripts.script_root
+        WHERE
+            execution_mode = 0 AND tag = ? AND
+            (consumed_block_num IS NULL OR consumed_block_num > ?) AND rowid >= ?
```

Comment:
You also need to exclude notes that were created _after_ the block number


```suggestion
        WHERE
            execution_mode = 0 AND tag = ?1 AND
            block_num <= ?2 AND
            (consumed_block_num IS NULL OR consumed_block_num > ?2) AND rowid >= ?3
```

Reply:
If possible, could you squeeze in a test case for this?